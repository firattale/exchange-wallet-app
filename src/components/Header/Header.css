.header-container {
    background-color: #387ad7;
    padding: 15px 15px;
    display: flex;
    justify-content: space-between;
}

.header-button,
.header-button:hover, 
.header-button:active, 
.header-button:focus {
    color: white;
    background-color: #387ad7 !important;
    border: 0;
    cursor: pointer;
    font-size: 16px;
    outline: none !important;
}
.header-button:disabled {
    opacity: 0.5;
    background-color: #387ad7;
}