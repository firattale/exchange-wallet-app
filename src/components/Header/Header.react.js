import React from 'react'
import './Header.css';
import ExchangeRates from '../ExchangeRates/ExchangeRates.react'
import { Button } from 'reactstrap';
import { updateWallet, selectFirstWalletError } from '../../app/walletSlice.js';
import { useSelector, useDispatch } from 'react-redux';

export const Header = ({ firstAmount, firstCurrency, secondCurrency, currencyRate }) => {
    const firstWalletAmount = useSelector(state => state.wallet[firstCurrency].toFixed(2));
    const secondWalletAmount = useSelector(state => state.wallet[secondCurrency].toFixed(2));
    const error = useSelector(selectFirstWalletError);

    const isDisabled = Boolean(+firstAmount > firstWalletAmount || firstAmount === "" || firstCurrency === secondCurrency || error);
    const dispatch = useDispatch();

    const handleExchange = () => {
        dispatch(updateWallet({
            firstWalletType: firstCurrency,
            secondWalletType: secondCurrency,
            firstAmount: +firstWalletAmount - firstAmount,
            secondAmount: +secondWalletAmount +(firstAmount * currencyRate),
        }))
    }
    return (
        <div className="header-container">
            <Button type="button" className="header-button">Cancel</Button>
            <ExchangeRates />
            <Button type="button" data-testid="exchangeButton" className="header-button" disabled={isDisabled} onClick={handleExchange} >Exchange</Button>
        </div>
    )
}

export default Header;