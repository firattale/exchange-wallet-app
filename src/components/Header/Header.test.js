import React from 'react';
import { mount } from 'enzyme';
import { Header } from './Header.react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from "../../app/walletSlice"



describe('Header', () => {
    const mockAppState = {
        exchange: {
            firstCurrency: "GBP",
            firstSign: "Â£",
            secondCurrency: "USD",
            currencyRate: "",
            secondSign: "$"
        },
        wallet: {
            USD: 40.00,
            GBP: 50.00,
            EUR: 60.00,
            firstAmount: "",
            secondAmount: "",
            firstWalletError: null
        }
    }
    const mockProps = {
        firstAmount: 50.00,
        firstCurrency: "GBP",
        secondCurrency: "USD",
        currencyRate: ""
    }

    const mockStore = createStore(reducer, mockAppState);
    mockStore.dispatch = jest.fn();
    const getWrapper = () => mount(
        <Provider store={mockStore}>
            <Header {...mockProps} />
        </Provider>
    );

    afterEach(() => {
        jest.clearAllMocks()
    });

    it('should render correctly in "debug" mode', () => {
        const wrapper = getWrapper();
        expect(wrapper).toMatchSnapshot();
    });

    it('should call dispatch func when  number inputs change', () => {

        const wrapper = getWrapper();
        const input = wrapper.find('[data-testid="exchangeButton"]').at(1);
        input.simulate("click");
        expect(mockStore.dispatch).toHaveBeenCalledTimes(2);
        expect(mockStore.dispatch).toHaveBeenCalledWith({ "payload": { "firstAmount": 0, "firstWalletType": "GBP", "secondAmount": 40, "secondWalletType": "USD" }, "type": "wallet/updateWallet" });
    });

});