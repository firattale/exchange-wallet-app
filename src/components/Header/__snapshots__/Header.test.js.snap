// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly in "debug" mode 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Header
    currencyRate=""
    firstAmount={50}
    firstCurrency="GBP"
    secondCurrency="USD"
  >
    <div
      className="header-container"
    >
      <Button
        className="header-button"
        color="secondary"
        tag="button"
        type="button"
      >
        <button
          aria-label={null}
          className="header-button btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
      </Button>
      <ExchangeRates>
        <div
          className="exchange-container"
        >
          <span
            className="currency"
          >
            Â£
          </span>
          1=
          <span
            className="currency"
          >
            $
          </span>
          <span
            className="currency"
          />
        </div>
      </ExchangeRates>
      <Button
        className="header-button"
        color="secondary"
        data-testid="exchangeButton"
        disabled={false}
        onClick={[Function]}
        tag="button"
        type="button"
      >
        <button
          aria-label={null}
          className="header-button btn btn-secondary"
          data-testid="exchangeButton"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Exchange
        </button>
      </Button>
    </div>
  </Header>
</Provider>
`;
